internal/repository/prune.go

```diff
@@ -385,12 +388,18 @@ func decidePackAction(ctx context.Context, opts PruneOptions, repo *Repository,
 			// if this is a data pack and --repack-cacheable-only is set => keep pack!
 			stats.Packs.Keep++
 
+		//wpl 2024-11-08 - repack when packSize is smaller than opts.SmallPackBytes
+		case opts.RepackSmall && (uint64(packSize) <= opts.SmallPackBytes):
```

Comment:
I don't think we need a special case here. Just set `targetPackSize` to `opts.SmallPackBytes` if the latter is non-zero and `opts.RepackSmall` is set.